README
NAME
    Dist::Zilla::PluginBundle::BAREFOOT - Dist::Zilla configuration the way
    BAREFOOT does it

VERSION
    version 0.03_03

SYNOPSIS
       # in dist.ini
       [@BAREFOOT]

DESCRIPTION
    This is a Dist::Zilla PluginBundle. It is roughly equivalent to the
    following dist.ini:

            ; version provider
            ; hopefully soemething here soon
 
            ; choose files to include
            [GatherDir]                                                     ; everything under top dir
            exclude_filename = README.pod           ; skip this generated file
            exclude_filename = META.json            ; skip this generated file
 
            ;[PruneCruft]                                           ; default stuff to skip
            ;[ManifestSkip]                                         ; if -f MANIFEST.SKIP, skip those, too
 
            ; this should probably be moved to metadata section
            [AutoPrereqs]                                           ; find prereqs from code
            skip = ^t::lib
 
            ; file modifications
            [OurPkgVersion]                                         ; add $VERSION = ... to all files
            ;[InsertCopyright                                       ; add copyright at "# COPYRIGHT"
            [PodWeaver]                                                     ; generate Pod
            config_plugin = @BAREFOOT                       ; allows Pod::WikiDoc and a few other bits and bobs
 
            ; generated files
            [License]                                                       ; boilerplate license
            [ReadmeAnyFromPod]                                      ; create README.pod in repo directory
            type = pod
            filename = README.pod
            location = root
 
            ; should this be in metadata section?
            [Bugtracker]
            web = http://github.com/me/%l/issues
 
            ; xt tests
            ;[Test::PodSpelling]                                    ; xt/author/pod-spell.t
            ;[MetaTests]                                                    ; xt/release/meta-yaml.t
            ;[PodSyntaxTests]                                       ; xt/release/pod-syntax.t
            ;[PodCoverageTests]                                     ; xt/release/pod-coverage.t
            ;[Test::Version]                                                ; xt/release/test-version.t
 
            ; metadata
            ;[MinimumPerl]                                          ; determine minimum perl version
            [GithubMeta]
            remote = origin
 
            ;[MetaYAML]                                                     ; generate META.yml (v1.4)
            [MetaJSON]                                                      ; generate META.json (v2)
 
            ;[MetaNoIndex]                                          ; sets 'no_index' in META
            ;directory = t
            ;directory = xt
            ;directory = examples
            ;directory = corpus
            ;package = DB                                           ; just in case
 
            ; can't get this one to work right ATM
            ; [MetaProvides::Package]                       ; add 'provides' to META files
            ; meta_noindex = 1                                      ; respect prior no_index directives
 
            ; build system
            [ExecDir]                                                       ; include 'bin/*' as executables
            [ShareDir]                                                      ; include 'share/' for File::ShareDir
            [MakeMaker]                                                     ; create Makefile.PL
 
            ; manifest (after all generated files)
            [Manifest]                                                      ; create MANIFEST
 
            ; before release
            [Git::Check]                                            ; ensure all files checked in
            allow_dirty = dist.ini
            allow_dirty = Changes
            allow_dirty = README.pod                        ; ignore this generated file
            allow_dirty = META.json                         ; ignore this generated file
 
            ;[CheckPrereqsIndexed]                          ; ensure prereqs are on CPAN
            [CheckChangesHasContent]                        ; ensure Changes has been updated
            ;[CheckExtraTests]                                      ; ensure xt/ tests pass
            [TestRelease]                                           ; ensure t/ tests pass
            [ConfirmRelease]                                        ; prompt before uploading
 
            ; releaser
            [UploadToCPAN]                                          ; uploads to CPAN
 
            ; after release
            [Git::Tag]                                                      ; tag repo with custom tag
            tag_format = v%v                                        ; this one is overridable
            tag_message = version %v for CPAN       ; this one isn't
 
            ; NextRelease acts *during* pre-release to write $VERSION and
            ; timestamp to Changes and  *after* release to add a new {{$NEXT}}
            ; section, so to act at the right time after release, it must come
            ; after UploadToCPAN but before Git::Commit in the dist.ini.  It
            ; will still act during pre-release as usual.
            [NextRelease]
 
            [Git::Commit]                                           ; commit Changes (for new dev)
 
            [Git::Push]                                                     ; push repo to remote
            push_to = origin

USAGE
    To use this PluginBundle, just add it to your dist.ini. You can provide
    the following options:

    *   "auto_prereq" -- This indicates whether AutoPrereq should be used or
        not. Default is 1.

    *   "tag_format" -- Given to "Git::Tag". Default is 'v%v'.

    *   "version_regexp" -- Given to "Git::NextVersion". Default is
        '^v(.+)$'.

    *   "git_remote" -- Given to "Git::Push" _in addition to_ origin; given
        to GithubMeta _instead of_ origin.

    *   "fake_release" -- Swaps FakeRelease for UploadToCPAN. Mostly useful
        for testing a dist.ini without risking a real release.

    *   "weaver_config" -- Specifies a Pod::Weaver bundle. Defaults to
        @BAREFOOT.

    *   "stopwords" -- Add stopword for Test::PodSpelling (can be repeated).

    *   "no_spellcheck" -- Omit Test::PodSpelling tests.

INSTALLATION
    If you want to make sure you have all the necessary prereqs, try this
    (from the dir you checked out the distro into):

            perl -lne 'print $1 if /Dependencies/../VERSION/ and /use\s+(\S+)/' lib/*/*/PluginBundle/BAREFOOT.pm | cpanm -n

SEE ALSO
    *   Dist::Zilla

    *   Dist::Zilla::Plugin::PodWeaver

AUTHOR
    Buddy Burden <barefoot@cpan.org>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2013 by Buddy Burden.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)

SUPPORT
  Perldoc
    You can find documentation for this module with the perldoc command.

      perldoc Dist::Zilla::PluginBundle::BAREFOOT

  Bugs / Feature Requests
    none



Changes
Revision history for Dist-Zilla-PluginBundle-BAREFOOT

0.03_05   2013-05-06 00:41:17 PST8PDT
	
	* fix failing tests when installing from CPAN (GitHub #9)
	* check for unbumped version (GitHub #12)

0.03_04   2013-05-02 01:58:51 PST8PDT

	* complete the transmogrification of dist.ini
		(to make self-hosting: GitHub #2)

0.03_03   2013-04-28 14:41:58 PST8PDT

	* made plugins in the bundle match the dist.ini
		(first step toward fixing GitHub #2)

0.03_02   2013-04-27 23:30:18 PST8PDT

	* fixed build error when doing `dzil release`

0.03_01   2013-02-25 02:08:35 PST8PDT

	* added check for uncommitted files
	* fixed double commit on release [GitHub #1]
	* ditching README.pod in favor of (static) README.md [GitHub #4]
	* adding dzil build dirs to Git ignore file

0.03      2013-02-21 12:47:46 PST8PDT

	*	added CheckChangesHasContent
	*	cleaned up Changes file

0.02      2013-02-21 12:36:12 PST8PDT

	*	added AutoPrereqs
		should fix the majority of the CPAN Testers failures

	*	change FakeRelease to UploadToCPAN

0.01      2012-12-26 14:07:42 America/Los_Angeles

	*	initial version
		shamelessly stolen from DAGOLDEN's plugin bundle
		(but with his permission)


lib/Pod/Weaver/PluginBundle/BAREFOOT.pm
use strict;
use warnings;

{
	package Pod::Weaver::PluginBundle::BAREFOOT;

	# Dependencies
	use Pod::Weaver 3.101635; # fixed ABSTRACT scanning
	use Pod::Weaver::Config::Assembler;

	use Pod::Weaver::Plugin::WikiDoc ();
	use Pod::Elemental::Transformer::List 0.101620 ();
	use Pod::Weaver::Section::Support 1.001 ();


	our $VERSION = '0.03_05'; # VERSION

	my $bugtracker_content = <<'END';
		This module is on GitHub.  Feel free to fork and submit patches.  Please note that I develop
		via TDD (Test-Driven Development), so a patch that includes a failing test is much more
		likely to get accepted (or least likely to get accepted more quickly).

		If you just want to report a problem or suggest a feature, that's okay too.  You can create
		an issue on GitHub here: {WEB}.
END


	sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

	sub mvp_bundle_config
	{
		my @plugins;
		push @plugins, (
			[ '@BAREFOOT/WikiDoc',     _exp('-WikiDoc'),	{} ],

			[ '@BAREFOOT/CorePrep',    _exp('@CorePrep'),	{} ],
			[ '@BAREFOOT/Name',        _exp('Name'),		{} ],
			[ '@BAREFOOT/Version',     _exp('Version'),		{
																format => "This document describes version %v of %m.",
															}
			],

			[ '@BAREFOOT/Synopsis',    _exp('Generic'),		{ header      => 'SYNOPSIS'    } ],
			[ '@BAREFOOT/Description', _exp('Generic'),		{ header      => 'DESCRIPTION' } ],
			[ '@BAREFOOT/Overview',    _exp('Generic'),		{ header      => 'OVERVIEW'    } ],
		);

		for my $plugin (
			[ 'Attributes', 			_exp('Collect'),	{ command => 'attr'   } ],
			[ 'Methods',    			_exp('Collect'),	{ command => 'method' } ],
			[ 'Functions',  			_exp('Collect'),	{ command => 'func'   } ],
		){
			$plugin->[2]->{'header'} = uc $plugin->[0];
			push @plugins, $plugin;
		}

		push @plugins, (
			[ '@BAREFOOT/Leftovers', _exp('Leftovers'),		{} ],
			[ '@BAREFOOT/Support',   _exp('Support'),		{
																perldoc				=> 1,
																websites			=> 'none',
																bugs				=> 'metadata',
																bugs_content		=> $bugtracker_content,
																repository_link		=> 'both',
																repository_content	=> 'none',
															}
			],
			[ '@BAREFOOT/Authors',   _exp('Authors'),		{} ],
			[ '@BAREFOOT/Legal',     _exp('Legal'),			{} ],
			[ '@BAREFOOT/List',      _exp('-Transformer'),	{ transformer => 'List' } ],
		);

		return @plugins;
	}

}

# ABSTRACT: BAREFOOT's default Pod::Weaver config
# COPYRIGHT

1;

__END__

=pod

=head1 NAME

Pod::Weaver::PluginBundle::BAREFOOT - BAREFOOT's default Pod::Weaver config

=head1 VERSION

This document describes version 0.03_05 of Pod::Weaver::PluginBundle::BAREFOOT.

=head1 DESCRIPTION

This is a L<Pod::Weaver> PluginBundle.  It is roughly equivalent to the
following weaver.ini:

   [-WikiDoc]
 
   [@CorePrep]
 
   [Name]
   [Version]
 
   [Generic / SYNOPSIS]
   [Generic / DESCRIPTION]
   [Generic / OVERVIEW]
 
   [Collect / ATTRIBUTES]
   command = attr
 
   [Collect / METHODS]
   command = method
 
   [Collect / FUNCTIONS]
   command = func
 
   [Leftovers]
   [Support]
   perldoc = 1
   websites = none
   bugs = metadata
   bugs_content = ... stuff (web only, email omitted) ...
   repository_link = both
   repository_content = none
 
   [Authors]
   [Legal]
 
   [-Transformer]
   transfomer = List

=for Pod::Coverage mvp_bundle_config

=head1 USAGE

This PluginBundle is used automatically with the CE<lt>@BAREFOOTE<gt> L<Dist::Zilla>
plugin bundle.

=head1 SEE ALSO

=over

=item *

L<Pod::Weaver>

=item *

L<Pod::Weaver::Plugin::WikiDoc>

=item *

L<Pod::Elemental::Transformer::List>

=item *

L<Dist::Zilla::Plugin::PodWeaver>

=back

=head1 AUTHOR

Buddy Burden <barefoot@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2013 by Buddy Burden.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)

=cut


lib/Pod/Weaver/Section/LegalWithAddendum.pm
use 5.012;
use MooseX::Declare;
use Method::Signatures::Modifiers;

class Pod::Weaver::Section::LegalWithAddendum extends Pod::Weaver::Section::Legal
{

	our $VERSION = '0.03_05'; # VERSION


	has addendum => ( is => 'ro', isa => 'Str', predicate => '_has_addendum' );

	override weave_section ($document, $input)
	{
		super();
		my $legal = $document->children->[-1];
		$legal->children->[0] .= "\n\n" . $self->addendum if $self->_has_addendum;
	}
}


1;


# ABSTRACT: Dist::Zilla configuration the way BAREFOOT does it
# COPYRIGHT

__END__

=pod

=head1 NAME

Pod::Weaver::Section::LegalWithAddendum - Dist::Zilla configuration the way BAREFOOT does it

=head1 VERSION

This document describes version 0.03_05 of Pod::Weaver::Section::LegalWithAddendum.

=head1 AUTHOR

Buddy Burden <barefoot@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2013 by Buddy Burden.

This is free software, licensed under:

  The Artistic License 2.0 (GPL Compatible)

=cut


Makefile.PL

use strict;
use warnings;

use 5.012;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla configuration the way BAREFOOT does it",
  "AUTHOR" => "Buddy Burden <barefoot\@cpan.org>",
  "BUILD_REQUIRES" => {
    "File::Temp" => 0,
    "Path::Class" => 0,
    "Test::Command" => "0.10",
    "Test::Most" => "0.25"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-BAREFOOT",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "NAME" => "Dist::Zilla::PluginBundle::BAREFOOT",
  "PREREQ_PM" => {
    "Dist::Zilla" => "4.3",
    "Dist::Zilla::Plugin::Bugtracker" => "1.102670",
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => "0.002",
    "Dist::Zilla::Plugin::CheckVersionIncrement" => "0.121750",
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => "0.10",
    "Dist::Zilla::Plugin::InsertCopyright" => "0.001",
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => "0.001008",
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => "0.120051",
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => "2.001002",
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::PluginBundle::Git" => "2.013",
    "List::MoreUtils" => 0,
    "Method::Signatures::Modifiers" => 0,
    "MooseX::Declare" => 0,
    "MooseX::Has::Sugar" => 0,
    "MooseX::Types::Moose" => 0,
    "Pod::Elemental::Transformer::List" => "0.101620",
    "Pod::Weaver" => "3.101635",
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Support" => "1.001",
    "Pod::Wordlist::hanekomu" => "1.113620",
    "autodie" => "2.00",
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.03_05",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



